corpus/transform.py

```diff
@@ -337,11 +339,94 @@ def run(self):
+        :return: Merged corpus.
+        """
+        # Merge speakers.
+        base_spk = {spk.name: spk for spk in base.top_level_speakers()}
+        to_add_spk = {spk.name: spk for spk in to_add.top_level_speakers()}
+        for name, spk in to_add_spk.items():
+            if name in base_spk:
+                assert spk.__dict__ == base_spk[name].__dict__, (
+                    f"Found same speaker {name} with different attributes in the base/to be added corpus.\n"
+                    f"Speaker attributes (base corpus): {base_spk[name]}.\n"
+                    f"Speaker attributes (corpus to be added): {spk}."
+                )
+            else:
+                base.add_speaker(spk)
+
+        # Merge recordings.
+        base_rec = {rec.name: rec for rec in base.top_level_recordings()}
+        add_rec = {rec.name: rec for rec in to_add.top_level_recordings()}
+        for name, rec in add_rec.items():
+            if name in base_rec:
+                MergeCorporaJob.merge_recordings(base_rec[name], rec)
+            else:
+                base.add_recording(rec)
+
+        # Merge subcorpora.
+        base_sc = {sc.name: sc for sc in base.top_level_subcorpora()}
+        add_sc = {sc.name: sc for sc in to_add.top_level_subcorpora()}
+        for name, sc in add_sc.items():
+            if name in base_sc:
+                MergeCorporaJob.merge_corpora(base_sc[name], sc)
+            else:
+                base.add_subcorpus(sc)
+
+        return base
+
+    @staticmethod
+    def merge_recordings(base: corpus.Recording, to_add: corpus.Recording):
+        """
+        Merges a given base recording with another recording.
+
+        :param base: Base recording.
+        :param to_add: Recording to be merged into :param:`base`.
+        """
+        assert base.audio == to_add.audio, "Same recording name points do different audio files"
+
+        # Merge the speakers.
+        base_spk = {spk.name: spk for spk in base.speakers}
+        to_add_spk = {spk.name: spk for spk in to_add.speakers}
+        for name, spk in to_add_spk.items():
+            if name in base_spk:
+                assert spk.__dict__ == base_spk[name].__dict__, (
+                    f"Found same speaker {name} with different attributes in the base/to be added corpus.\n"
+                    f"Speaker attributes (base corpus): {base_spk[name]}.\n"
+                    f"Speaker attributes (corpus to be added): {spk}."
+                )
+            else:
+                base.speakers[name] = spk
+
+        # Merge the segments.
+        base_seg = {seg.name: seg for seg in base.segments}
+        add_seg = {seg.name: seg for seg in to_add.segments}
+        for name, seg in add_seg.items():
+            assert name not in base_seg, (
+                "The same segment exists in two corpora. "
+                "The MergeCorporaJob can't handle this situation, even if both segments are equal.\n"
```

Comment:
> even if both segments are equal.

mh.. should we implement this?

Reply:
I can also assert that `seg.__dict__` == `base_seg[name].__dict__`, just as I did for the speakers.

Reply:
I've implemented the dict check. I've made sure to remove the recording, since two segments would always be different if we compared them along with their associated recordings (mutable object).